service/history/hsm/tree.go

```diff
@@ -419,3 +413,34 @@ func (c Collection[T]) Transition(stateMachineID string, transitionFn func(T) (T
 	}
 	return MachineTransition(node, transitionFn)
 }
+
+// GenerateEventLoadToken generates a token for loading a history event from an [Environment].
+// Must be called within an [Environment.Access] function block with either read or write access.
```

Comment:
Does this restriction still apply since this function is no longer a method on `NodeBackend`?

Reply:
Thanks, I forgot to update this. I think history events are treated as immutable so they don't have to be cloned when accessed outside of the lock.